Sources/TSCUtility/SQLite.swift

```diff
@@ -15,6 +15,100 @@ import TSCBasic
 
 /// A minimal SQLite wrapper.
 public struct SQLite {
+    /// The location of the database.
+    public let location: Location
+
+    /// The configuration for the database.
+    public let configuration: Configuration
+
+    /// Pointer to the database.
+    let db: OpaquePointer
+
+    /// Create or open the database at the given path.
+    ///
+    /// The database is opened in serialized mode.
+    public init(location: Location, configuration: Configuration? = nil) throws {
+        self.location = location
+        self.configuration = configuration ?? Configuration()
+
+        var db: OpaquePointer?
+        try Self.checkError("unable to open database at \(self.location)") {
+            sqlite3_open_v2(
+                location.pathString,
+                &db,
+                SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FULLMUTEX,
+                nil
+            )
+        }
+
+        self.db = db!
+        try Self.checkError { sqlite3_extended_result_codes(db, 1) }
```

Comment:
this code has no changed, do you think we should add more explicit error?

Reply:
nice but not a must

Reply:
It would be good to add more context here, but not needed as part of this PR.  I suspect we'll want to do a second pass at this code and make the unchanged parts more robust at some point.